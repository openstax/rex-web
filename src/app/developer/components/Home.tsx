import React from 'react';
import styled from 'styled-components/macro';
import Footer from '../../components/Footer';
import Layout, { LayoutBody } from '../../components/Layout';
import { H1, textRegularStyle } from '../../components/Typography';
import { contentWrapperMaxWidth } from '../../content/components/constants';
import DisplayNotifications from '../../notifications/components/Notifications';
import Books from './Books';
import Notifications from './Notifications';
import Routes from './Routes';

// tslint:disable-next-line:variable-name
const HomeStyle = styled.div`
  ${textRegularStyle}
  max-width: ${contentWrapperMaxWidth}rem;
  margin: 0 auto;
  padding: 2rem 0;
  flex: 1;
`;

// tslint:disable-next-line:variable-name
const Row = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
`;

// tslint:disable-next-line:variable-name
const Col = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`;

// tslint:disable-next-line:variable-name
const Wrapper = styled(LayoutBody)`
  flex: 1;
`;

// tslint:disable-next-line:variable-name
const Home: React.SFC = () => <Layout>
  <DisplayNotifications />
  <Wrapper>
    <HomeStyle>
      <H1>REX Developer Homepage</H1>
      <Row>
        <Col>
          <Notifications />
          <Routes />
        </Col>
        <Col>
          <Books />
        </Col>
      </Row>
    </HomeStyle>
  </Wrapper>
  <Footer />
</Layout>;

export default Home;
