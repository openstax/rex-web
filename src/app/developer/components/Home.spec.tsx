import React from 'react';
import renderer from 'react-test-renderer';
import createTestServices from '../../../test/createTestServices';
import { book } from '../../../test/mocks/archiveLoader';
import TestContainer from '../../../test/TestContainer';
import Home from './Home';

jest.mock('../../../config.books', () => ({
  'some-id': {
    defaultVersion: '1.0',
  },
  'some-id-2': {
    defaultVersion: '1.0',
  },
}));

describe('Home', () => {
  const services = createTestServices();
  services.archiveLoader.mockBook({ ...book, id: 'some-id', title: 'Should be second' } as any);
  services.archiveLoader.mockBook({ ...book, id: 'some-id-2', title: '0 Should be first' } as any);

  it('matches snapshot and books are sorted', async() => {
    jest.spyOn(Date.prototype, 'getFullYear').mockReturnValue(2021);
    const component = renderer.create(<TestContainer services={services}>
      <Home />
    </TestContainer>);

    // defer promises...
    await new Promise((resolve) => setTimeout(resolve, 1));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
