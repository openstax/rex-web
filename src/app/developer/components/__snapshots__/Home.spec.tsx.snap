// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot and books are sorted 1`] = `
Array [
  <a
    className="HiddenLink-wwwegq-0 kZsmVR"
    href="#main-content"
    onClick={[Function]}
  >
    Skip to Content
  </a>,
  <a
    className="HiddenLink-wwwegq-0 kZsmVR"
    href={
      Array [
        "https://openstax.org/accessibility-statement",
      ]
    }
  >
    Go to accessibility page
  </a>,
  <div
    className="styled__BarWrapper-sc-3syvnw-12 ePgITh"
    data-analytics-region="openstax-navbar"
  >
    <div
      className="styled__TopBar-sc-3syvnw-0 iVnlSK"
      data-testid="navbar"
    >
      <a
        href="/"
      >
        <img
          alt="OpenStax Logo"
          className="styled__HeaderImage-sc-3syvnw-1 gTGoMI"
          role="img"
          src="logo.svg"
        />
      </a>
    </div>
  </div>,
  <div
    className="Layout__LayoutBody-s6mgs9-0 Home__Wrapper-kkxtzr-1 hvoruj"
  >
    <div
      className="Home__HomeStyle-kkxtzr-0 ebFfeP"
    >
      <h1
        className="headings__H1-sc-1xfiggv-0 cErBHT"
      >
        REX Developer Homepage
      </h1>
      <div>
        <h2
          className="headings__H2-sc-1xfiggv-1 hrMREQ"
        >
          Books
        </h2>
        <div
          className="Panel__Wrappper-sc-13xpmbe-0 dMnuSy"
        >
          <ul>
            <li>
              <h3
                className="headings__H3-sc-1xfiggv-2 dxSanG"
              >
                <a
                  className="ContentLink__StyledContentLink-ctb7jh-0 jAbAdk"
                  href="books/book-slug-1/pages/1-introduction-to-science-and-the-realm-of-physics-physical-quantities-and-units"
                  onClick={[Function]}
                >
                  0 Should be first
                </a>
              </h3>
            </li>
            <li>
              <h3
                className="headings__H3-sc-1xfiggv-2 dxSanG"
              >
                <a
                  className="ContentLink__StyledContentLink-ctb7jh-0 jAbAdk"
                  href="books/book-slug-1/pages/1-introduction-to-science-and-the-realm-of-physics-physical-quantities-and-units"
                  onClick={[Function]}
                >
                  Should be second
                </a>
              </h3>
            </li>
          </ul>
        </div>
      </div>
      <div>
        <h2
          className="headings__H2-sc-1xfiggv-1 hrMREQ"
        >
          Notifications
        </h2>
        <div
          className="Panel__Wrappper-sc-13xpmbe-0 dMnuSy"
        >
          <div
            className="Button__ButtonGroup-ayg7nk-1 jzhcUT"
          >
            <button
              className="Button-ayg7nk-0 bmzHbc"
              onClick={[Function]}
            >
              accept cookies
            </button>
            <button
              className="Button-ayg7nk-0 bmzHbc"
              data-testid="trigger-updates-available"
              onClick={[Function]}
            >
              update available
            </button>
            <button
              className="Button-ayg7nk-0 bmzHbc"
              data-testid="trigger-messages"
              onClick={[Function]}
            >
              app messages
            </button>
            <button
              className="Button-ayg7nk-0 bmzHbc"
              data-testid="trigger-inline-error"
              onClick={[Function]}
            >
              inline error
            </button>
            <button
              className="Button-ayg7nk-0 bmzHbc"
              data-testid="trigger-modal-error"
              onClick={[Function]}
            >
              modal error
            </button>
          </div>
        </div>
      </div>
      <div>
        <h2
          className="headings__H2-sc-1xfiggv-1 hrMREQ"
        >
          Routes
        </h2>
        <div
          className="Panel__Wrappper-sc-13xpmbe-0 dMnuSy"
        >
          <div>
            <h3
              className="headings__H3-sc-1xfiggv-2 dxSanG"
            >
              Developer Home
            </h3>
            path: 
            /
            <br />
            /books/list
            <br />
          </div>
          <div>
            <h3
              className="headings__H3-sc-1xfiggv-2 dxSanG"
            >
              Content
            </h3>
            path: 
            /books/:book_uuid([\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12})@:book_version/pages/:page_uuid([\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12})
            <br />
            /books/:book_uuid([\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12})@:book_version/pages/:page_slug
            <br />
            /books/:book_slug@:book_version/pages/:page_uuid([\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12})
            <br />
            /books/:book_slug@:book_version/pages/:page_slug
            <br />
            /books/:book_slug/pages/:page_uuid([\\da-z]{8}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{4}-[\\da-z]{12})
            <br />
            /books/:book_slug/pages/:page_slug
            <br />
          </div>
          <div>
            <h3
              className="headings__H3-sc-1xfiggv-2 dxSanG"
            >
              NotFound
            </h3>
            path: 
            /(.*)
            <br />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="styled__FooterWrapper-y5hgq4-4 gprQGv"
    data-analytics-region="footer"
  >
    <div
      className="styled__InnerFooter-y5hgq4-5 cVQewI"
    >
      <div
        className="styled__FooterTop-y5hgq4-6 emNNEw"
      >
        <div
          className="styled__TopBoxed-y5hgq4-7 jVeZhX"
        >
          <div
            aria-level={2}
            className="styled__Heading-y5hgq4-8 kYpDrD"
            role="heading"
          >
            Our mission is to improve educational access and learning for everyone.
          </div>
          <div
            className="styled__Mission-y5hgq4-9 hHuCFj"
            dangerouslySetInnerHTML={
              Object {
                "__html": "
  OpenStax is part of Rice University, which is a 501(c)(3) nonprofit.
  <a href=\\"/give\\" target=\\"_blank\\">Give today</a> and help us reach more students.
",
              }
            }
          />
          <div
            className="styled__Column1-y5hgq4-12 bBGxDm"
          >
            <div
              className="styled__ColumnHeading-y5hgq4-15 iXuffB"
            >
              Help
            </div>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/contact"
              rel=""
              target="_self"
            >
              Contact Us
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="https://openstax.secure.force.com/help"
              rel="noopener"
              target="_blank"
            >
              Support Center
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/faq"
              rel=""
              target="_self"
            >
              FAQ
            </a>
          </div>
          <div
            className="styled__Column2-y5hgq4-13 gOHYYF"
          >
            <div
              className="styled__ColumnHeading-y5hgq4-15 iXuffB"
            >
              OpenStax
            </div>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/press"
              rel=""
              target="_self"
            >
              Press
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="http://www2.openstax.org/l/218812/2016-10-04/lvk"
              rel="noopener"
              target="_blank"
            >
              Newsletter
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/careers"
              rel=""
              target="_self"
            >
              Careers
            </a>
          </div>
          <div
            className="styled__Column3-y5hgq4-14 tBdZW"
          >
            <div
              className="styled__ColumnHeading-y5hgq4-15 iXuffB"
            >
              Policies
            </div>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/accessibility-statement"
              rel=""
              target="_self"
            >
              Accessibility Statement
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/tos"
              rel=""
              target="_self"
            >
              Terms of Use
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/license"
              rel=""
              target="_self"
            >
              Licensing
            </a>
            <a
              className="styled__FooterLink-y5hgq4-10 ZbGzz"
              href="/privacy-policy"
              rel=""
              target="_self"
            >
              Privacy Policy
            </a>
          </div>
        </div>
      </div>
      <div
        className="styled__FooterBottom-y5hgq4-16 DOTyv"
      >
        <div
          className="styled__BottomBoxed-y5hgq4-17 hViXPP"
        >
          <div
            className="styled__Copyrights-y5hgq4-18 bCRniQ"
            dangerouslySetInnerHTML={
              Object {
                "__html": "
  <div data-html=\\"copyright\\">Â© 1999-2021, Rice University. Except where otherwise noted, textbooks on this site
    are licensed under a
    <a href=\\"https://creativecommons.org/licenses/by/4.0/\\" target=\\"_blank\\"
    rel=\\"noreferrer\\">Creative Commons Attribution 4.0 International License</a>.
  </div>
  <div data-html=\\"apStatement\\">
    Advanced Placement<sup>Â®</sup> and AP<sup>Â®</sup> are trademarks registered and/or owned by the College Board,
    which is not affiliated with, and does not endorse, this site.
  </div>
",
              }
            }
          />
          <div
            className="styled__Social-y5hgq4-19 kjjZEm"
            role="directory"
          >
            <a
              aria-label="facebook"
              className="styled__SocialIcon-y5hgq4-20 hUKzFZ"
              href="https://www.facebook.com/openstax"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="styled__FBIcon-y5hgq4-0 lbjeVE StyledIconBase-sc-bdy9j4 cCvWXa"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 320 512"
              >
                <path
                  d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              aria-label="twitter"
              className="styled__SocialIcon-y5hgq4-20 hUKzFZ"
              href="https://twitter.com/openstax"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="styled__TwitterIcon-y5hgq4-1 dMlpbg StyledIconBase-sc-bdy9j4 cCvWXa"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 512 512"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              aria-label="linkedin"
              className="styled__SocialIcon-y5hgq4-20 hUKzFZ"
              href="https://www.linkedin.com/company/openstax"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="styled__LinkedInIcon-y5hgq4-3 hvjOHs StyledIconBase-sc-bdy9j4 cCvWXa"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 448 512"
              >
                <path
                  d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              aria-label="instagram"
              className="styled__SocialIcon-y5hgq4-20 hUKzFZ"
              href="https://www.instagram.com/openstax/"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                className="styled__IGIcon-y5hgq4-2 hQnbtn StyledIconBase-sc-bdy9j4 cCvWXa"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 448 512"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              className="styled__BottomLink-y5hgq4-11 iKPkVp"
              href="http://www.rice.edu"
              rel="noopener"
              target="_blank"
            >
              <img
                alt="Rice University logo"
                className="styled__FooterLogo-y5hgq4-21 NqIwN"
                src="rice-white-text.png"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>,
]
`;
